common/djangoapps/third_party_auth/middleware.py

```diff
@@ -23,3 +25,46 @@ def get_redirect_uri(self, request, exception):
             redirect_uri = pipeline.AUTH_DISPATCH_URLS[auth_entry]
 
         return redirect_uri
+
+
+class ResetSessionIfPipelineBrokenMiddleware(object):
+    """
+    Middleware signs the user out if they need to provide data sharing consent,
+    haven't, and left the TPA pipeline prematurely
+    """
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        """
+        Conditionally sign out users who don't provide data sharing consent
+        """
+        allowed_module_prefixes = (
```

Comment:
@haikuginger 

> How would you feel about making this list modifiable via a setting?

I would feel bad, because it would impose additional burden on devops to keep the setting up to date.

I might have miscommunicated this - the idea is to check pipeline first, than fallback to curated list of other modules. So, I'm not suggesting to replace this list of allowed modules entirely, but to add another condition to let any method that is already on the pipeline (but not in the module listed here) to proceed.

There's a caveat of that method calling other methods, but we'll have to communicate that in "a method to add hooks into pipeline".

Actually, I'm not feeling really convinced module whitelist is a good approach here, but I haven't  come up with a better one.